### PR TITLE
[gradle-plugin] Support frameworks and extra options in the new plugin

### PR DESCRIPTION
**Pull request by ilmat192**

### CODE REVIEW

File: tools/kotlin-native-gradle-plugin/src/main/kotlin/org/jetbrains/kotlin/gradle/plugin/experimental/internal/KotlinNativeLibraries.kt [hunk 1]
```diff
@@ -68,3 +69,49 @@ open class KotlinNativeKLibraryImpl @Inject constructor(
 
     override val outputRootName = "lib"
 }
+
+open class KotlinNativeKLibraryImpl @Inject constructor(
```
Comment (olonho) [inline tools/kotlin-native-gradle-plugin/src/main/kotlin/org/jetbrains/kotlin/gradle/plugin/experimental/internal/KotlinNativeLibraries.kt]: KotlinNativeLibraryImpl?

File: tools/kotlin-native-gradle-plugin/src/main/kotlin/org/jetbrains/kotlin/gradle/plugin/experimental/internal/KotlinNativeMainComponent.kt [hunk 2]
```diff
@@ -82,5 +82,8 @@ open class KotlinNativeMainComponent @Inject constructor(
 
         @JvmStatic
         val KLIBRARY = OutputKind.KLIBRARY
+
+        @JvmStatic
+        val FRAMEWORK = OutputKind.FRAMEWORK
```
Comment (olonho) [inline tools/kotlin-native-gradle-plugin/src/main/kotlin/org/jetbrains/kotlin/gradle/plugin/experimental/internal/KotlinNativeMainComponent.kt]: What about static and dynamic libs as well?

File: tools/kotlin-native-gradle-plugin/src/main/kotlin/org/jetbrains/kotlin/gradle/plugin/experimental/internal/OutputKind.kt [hunk 3]
```diff
@@ -3,28 +3,46 @@ package org.jetbrains.kotlin.gradle.plugin.experimental.internal
...
+        2,
+        null,
+        KotlinNativeUsage.FRAMEWORK,
+        false
+    ) {
+        override fun availableFor(target: KonanTarget) =
+            target.family == Family.OSX || target.family == Family.IOS
+    };
+
+    open fun availableFor(target: KonanTarget) = true
```
Comment (vvlevchenko) [inline tools/kotlin-native-gradle-plugin/src/main/kotlin/org/jetbrains/kotlin/gradle/plugin/experimental/internal/OutputKind.kt]: nice

File: tools/kotlin-native-gradle-plugin/src/main/kotlin/org/jetbrains/kotlin/gradle/plugin/experimental/internal/KotlinNativeMainComponent.kt (hunk 2 above)
Comment (ilmat192) [reply tools/kotlin-native-gradle-plugin/src/main/kotlin/org/jetbrains/kotlin/gradle/plugin/experimental/internal/KotlinNativeMainComponent.kt]: Adding them is not very complicated but I think that it's better to focus on integration with Kotlin MPP at the moment.

File: tools/kotlin-native-gradle-plugin/src/main/kotlin/org/jetbrains/kotlin/gradle/plugin/experimental/internal/KotlinNativeLibraries.kt (hunk 1 above)
Comment (ilmat192) [reply tools/kotlin-native-gradle-plugin/src/main/kotlin/org/jetbrains/kotlin/gradle/plugin/experimental/internal/KotlinNativeLibraries.kt]: Fixed

**Pull request merged**